src/Tools/AI Test Toolkit/src/TestSuite/AITTestSuite.Page.al

```diff
@@ -0,0 +1,371 @@
+                    Editable = false;
+                    ApplicationArea = All;
+                    Style = Favorable;
+                }
+                field("No. of Tests Failed"; Rec."No. of Tests Executed" - Rec."No. of Tests Passed")
+                {
+                    Editable = false;
+                    ApplicationArea = All;
+                    Style = Unfavorable;
+                    Caption = 'No. of Tests Failed';
+                    ToolTip = 'Specifies the number of tests failed in the current version.';
+
+                    trigger OnDrillDown()
+                    var
+                        AITLogEntries: Record "AIT Log Entry";
+                        AITLogEntry: Page "AIT Log Entries";
+                    begin
+                        AITLogEntries.SetFilterForFailedTestProcedures();
+                        AITLogEntries.SetRange("AIT Code", Rec.Code);
+                        AITLogEntries.SetRange(Version, Rec.Version);
+                        AITLogEntry.SetTableView(AITLogEntries);
+                        AITLogEntry.Run();
+                    end;
+                }
+                field("No. of Operations"; Rec."No. of Operations")
+                {
+                    ApplicationArea = All;
+                    Caption = 'No. of Operations';
+                    ToolTip = 'Specifies the number of operations executed in the current version.';
+                }
+                field("Total Duration"; this.TotalDuration)
+                {
+                    ApplicationArea = All;
+                    Editable = false;
+                    Caption = 'Total Duration';
+                    ToolTip = 'Specifies the Total Duration for executing all the selected tests in the current version.';
+                }
+                field("Average Duration"; this.AvgTimeDuration)
+                {
+                    Editable = false;
+                    ApplicationArea = All;
+                    Caption = 'Average Duration';
+                    ToolTip = 'Specifies the average time (ms) taken by the tests in the last run.';
+                }
+            }
+
+        }
+    }
+    actions
+    {
+        area(Processing)
+        {
+            action(Start)
+            {
+                Enabled = (this.EnableActions and (Rec.Status <> Rec.Status::Running));
+                ApplicationArea = All;
+                Caption = 'Start';
+                Image = Start;
+                Promoted = true;
+                PromotedOnly = true;
+                PromotedCategory = Process;
+                ToolTip = 'Starts running the AIT Suite.';
+
+                trigger OnAction()
+                begin
+                    CurrPage.Update(false);
+                    Rec.Find();
+                    if Rec."Input Dataset" = '' then
+                        Error('Please specify a dataset before starting the suite.');
+                    this.AITTestSuiteCU.ValidateDatasets(Rec);
```

Comment:
As discussed on the TechDays: please remove "this" for global variables and procedures.

Reply:
We are having internal discussion if we want to apply this. everywhere in the Microsoft owned code. We will follow up with the partner channel.

Reply:
I did comment the opposite, to add `this.` since I saw it in many places but not everywhere.
I think this discussion is necessary, but either way, it has to be used consistently.

Reply:
I have removed this for now we need to agree internally on usage of this.